### PR TITLE
Add ignore command (from #1257)

### PR DESCRIPTION
**Pull request by erivas**
Tried to follow implementation outline in #1257 , where this was labeled as good-first-issue/help-wanted.

### CODE REVIEW

**Comment by gsingh93** [comment]
Thanks! I'll test this out myself, but overall the implementation looks good. Would you also be able to add a test for this command? I think we should test different valid and invalid variants of `ignore`, where there are no breakpoints, where a non-existent breakpoint number is entered, etc.

File: pwndbg/commands/ignore.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+parser.add_argument("bpnum", type=int, default=None, nargs="?", help="The breakpoint number N.")
+parser.add_argument("count", type=int, help="The number to set COUNT.")
+
+
+@pwndbg.commands.ArgparsedCommand(parser)
+def ignore(bpnum, count):
+    if bpnum is None:
+        bpnum = max((bp.number for bp in gdb.breakpoints()), default=None)
+
+    bp = next((bp for bp in gdb.breakpoints() if bp.number == bpnum), None)
```
Comment (disconnect3d) [inline pwndbg/commands/ignore.py]: How about:

1. Changing this code to:
```py
if bpnum is None:
    bp = max(gdb.breakpoints(), key=lambda bp: bp.number, default=None)
else:
    bp = next((bp for bp in gdb.breakpoints() if bp.number == bpnum), None)
```

2. Changing the UX to match the original one a bit more:
```
pwndbg> help ignore
Set ignore-count of breakpoint number N to COUNT.
Usage is `ignore N COUNT'.

pwndbg> ignore 1 123
Will ignore next 123 crossings of breakpoint 1.

pwndbg> ignore 123 123
No breakpoint number 123.
```
So:
- we can show an example in the description
- display error as 'No breakpoint number <bpnum>.' if it is defined or 'No breakpoints set' if it is not
- for success path, inform user that we did it via "Will ignore next 123 crossings of breakpoint 1."

Comment (disconnect3d) [reply pwndbg/commands/ignore.py]: Fwiw for a multiline description with examples etc. you can use RawDescriptionHelpFormatter as we do in vmmap command: https://github.com/pwndbg/pwndbg/blob/dev/pwndbg/commands/vmmap.py#L50

Also I am wondering if we can grab last bp by just gdb.breakpoints()[-1] but idk if that is always the case so its safer to do it with max(..)

Comment (disconnect3d) [reply pwndbg/commands/ignore.py]: Hm, so we could have:
```py
bps = gdb.breakpoints()

if not bps:
    print(message.error("No breakpoints set."))

if bpnum is None:
    bp = max(bps, key=lambda bp: bp.number)
else:
    bp = next((bp for bp in bps if bp.number == bpnum), None)

    if bp is None:
        print(message.error("No breakpoint number %d." % bpnum))
        return

bp.ignore_count = count
print("Will ignore next %d crossings of breakpoint %d." % (count, bpnum))
```

**Comment by erivas** [comment]
> Thanks! I'll test this out myself, but overall the implementation looks good. Would you also be able to add a test for this command? I think we should test different valid and invalid variants of `ignore`, where there are no breakpoints, where a non-existent breakpoint number is entered, etc.

Thanks, added a few test cases in a file `test_command_ignore.py`, trying to follow how other commands are tested.

File: pwndbg/commands/ignore.py (hunk 1 above)
Comment (erivas) [reply pwndbg/commands/ignore.py]: Thanks! Applied the code changes.

I didn't add an example in the description, as an `usage: ...` line was already generated. It looks like this for me:
```
pwndbg> help ignore
usage: ignore [-h] [N] COUNT

Set ignore-count of breakpoint number N to COUNT.

While the ignore count is positive, execution will not stop on the breakpoint.

By default, if `N' is ommitted, the last breakpoint (i.e. greatest breakpoint number) will be used.

positional arguments:
  N           The breakpoint number N.
  COUNT       The number to set COUNT.

options:
  -h, --help  show this help message and exit
```

**Comment by gsingh93** [comment]
The tests look great! Any reason you're calling `gdb.execute("file " + REFERENCE_BINARY)` instead of `start_binary(REFERENCE_BINARY)` in some of the tests?

**Comment by disconnect3d** [comment]
> The tests look great! Any reason you're calling gdb.execute("file " + REFERENCE_BINARY) instead of start_binary(REFERENCE_BINARY) in some of the tests?

Seems like an optimization - you don't always have to start the binary but can only load it to gdb

**Pull request merged**